pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -228,6 +246,8 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) e
 
 // AppendInstanceHandler implements a service instance catalog operation
 func (c *Controller) AppendInstanceHandler(f func(*model.ServiceInstance, model.Event)) error {
+	c.storeLock.RLock()
+	defer c.storeLock.RUnlock()
```

Comment:
This method is called when the aggregate controller is initialized. So the lock in here seems unnecessary. The same goes with the Run() method, and the lock is not applied there. Not sure why you chose not to do it for the Run() method.
I think that some comments about not using the lock may be sufficient.

Reply:
This is for safety, AppendInstanceHandler can potentially be called from different go routine or during the time when list of registry being updated. So I think lock here is valid, o the other hand locking before calling Run, does not make too much sense to me, unless I am missing something. Could you please clarify why you think I should Lock before calling Run?